applications/ChimeraApplication/custom_strategies/strategies/fs_strategy_for_chimera.h

```diff
@@ -0,0 +1,836 @@
+            for(const auto& slave_dof : slave_dofs)
+            {
+                const auto slave_node_id = slave_dof->Id(); // DOF ID is same as node ID
+                auto& r_slave_node = rModelPart.Nodes()[slave_node_id];
+                r_slave_node.FastGetSolutionStepValue(FRACT_VEL_X)=0;
+                r_slave_node.FastGetSolutionStepValue(FRACT_VEL_Y)=0;
+            }
+        }
+
+        for(const auto& constraint : r_constraints_container)
+        {
+            const auto& master_dofs = constraint.GetMasterDofsVector();
+            const auto& slave_dofs = constraint.GetSlaveDofsVector();
+            ModelPart::MatrixType r_relation_matrix;
+            ModelPart::VectorType r_constant_vector;
+            constraint.CalculateLocalSystem(r_relation_matrix,r_constant_vector,rModelPart.GetProcessInfo());
+
+            IndexType slave_i = 0;
+            for(const auto& slave_dof : slave_dofs)
+            {
+                const auto slave_node_id = slave_dof->Id(); // DOF ID is same as node ID
+                auto& r_slave_node = rModelPart.Nodes()[slave_node_id];
+                IndexType master_j = 0;
+                for(const auto& master_dof : master_dofs)
+                {
+                    const auto master_node_id = master_dof->Id();
+                    const double weight = r_relation_matrix(slave_i, master_j);
+                    auto& r_master_node = rModelPart.Nodes()[master_node_id];
+
+                    r_slave_node.GetValue(FRACT_VEL) +=(r_master_node.FastGetSolutionStepValue(FRACT_VEL))*weight;
+
+                    ++master_j;
+                }
+                ++slave_i;
+            }
+        }
+
+        rModelPart.GetCommunicator().AssembleNonHistoricalData(FRACT_VEL);
+
+        for (typename ModelPart::NodeIterator itNode = rModelPart.NodesBegin(); itNode != rModelPart.NodesEnd(); itNode++)
+        {
+            array_1d<double,3>& rDeltaVel = itNode->GetValue(FRACT_VEL);
+            if ( rDeltaVel[0]*rDeltaVel[0] + rDeltaVel[1]*rDeltaVel[1] + rDeltaVel[2]*rDeltaVel[2] != 0.0)
+            {
+                itNode->FastGetSolutionStepValue(FRACT_VEL) = itNode->GetValue(FRACT_VEL);
+                rDeltaVel = array_1d<double,3>(3,0.0);
+            }
+        }
+    }
+
+
+    ///@}
+    ///@name Protected  Access
+    ///@{
+
+
+    ///@}
+    ///@name Protected Inquiry
+    ///@{
+
+
+    ///@}
+    ///@name Protected LifeCycle
+    ///@{
+
+
+    ///@}
+
+private:
+    ///@name Static Member Variables
+    ///@{
+
+
+    ///@}
+    ///@name Member Variables
+    ///@{
+    const double mAreaTolerance=1E-12;
+
+    ///@}
+    ///@name Private Operators
+    ///@{
+
+
+    ///@}
+    ///@name Private Operations
+    ///@{
+
+
+    void SetHoleVariablesToZero(ModelPart& rModelPart)
+    {
+        const array_1d<double,3> Zero(3,0.0);
+        for ( auto itElem = rModelPart.Elements().ptr_begin(); itElem != rModelPart.Elements().ptr_end(); ++itElem )
```

Comment:
I think that you can do this loop in parallel if you use the lock of the nodes when setting the values.